openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -38,15 +41,44 @@ def _field_values(fields, xblock):
     return result
 
 
+def _has_assignment_blocks(item):
+    """
+    Check if a given block contains children that are assignments.
+    Assignments have graded, has_score and nonzero weight attributes.
+    """
+    def leaf_filter(block):
+        # Filter for traversals to find leaf blocks
+        return (
+            block.location.block_type not in ('chapter', 'sequential', 'vertical') and
+            len(block.get_children()) == 0
```

Comment:
You're not using `self.get_children` here?

Reply:
it is being passed to the traversal since it takes a function that can return the children of a block as an argument